### ISSUE TITLE
1.0.0-ALPHA-5 doesn't generate native and nativeValue for StateFlow

### ISSUE THREAD

**Issue by Pschsch** [rickclephas/KMP-NativeCoroutines#95]

**Comment by Pschsch** [comment]
Need to use 0.13.3 and Kotlin 1.8.0 instead

**Comment by rickclephas** [comment]
Hi the 1.0.0-ALPHA releases work a little different from the older 0.x once.
Could you share some Kotlin code and what exactly it is you are missing?

**Comment by Pschsch** [comment]
`class SomeViewModel {

    sealed class State {
        object Loading : State()
        object Loaded : State()
    }

    private val _state = MutableStateFlow<State>(State.Loading)
    val state: StateFlow<State>
        get() = _state.asStateFlow()

}`
In 0.13.3 release, i have access to stateNative and stateNativeValue fields on iOS side.
But on 1.0.0-ALPHA-5 there are no these fields)

**Comment by rickclephas** [comment]
v1.0.0 is using KSP to generate the native properties.
For that you'll need to annotate your declarations with `@NativeCoroutines`.
Also see the [migration steps](

**Comment by Pschsch** [comment]
Have followed all migration steps. But still no native values. Now, it just hides property annotated with NativeCoroutines annotation (cause of Kotlin's HidesFronObjC annotation).
Code sample now:
```
class SomeViewModel {

    sealed class State {
        object Loading : State()
        object Loaded : State()
    }

    private val _state = MutableStateFlow<State>(State.Loading)
    
    @NativeCoroutines
    val state: StateFlow<State>
        get() = _state.asStateFlow()

}
```

**Comment by rickclephas** [comment]
Yeah, you'll need to apply the KSP plugin as well.
Is this `SomeViewModel` class inside your final KMM module (or is it part of a dependency)? If it's part of a dependency, you'll have to export that dependency as well.

**Comment by msa1422** [comment]
I have faced the same problem. I have applied ksp plugin, used the annotation, but I am unable to see the native StateFlow on the iOS side. In fact, the StateFlow marked with ```@NativeCoroutines``` is not accessible on iOS side at all, not even the normal declaration. I believe my ksp set is fine because it works with Android Room.
I will try to share a project where this issue can be reproduced

**Comment by msa1422** [comment]
As promised [here]() is the project where issue can be reproduced.

 The following ```build.gradle``` files use KSP and NativeCoroutines
[petsearch-shared:core:util](https://github.com/msa1422/KMM-Arch-PetSearch/blob/master/petsearch-shared/core/util/build.gradle.kts)
[petsearch-shared:ui:home](https://github.com/msa1422/KMM-Arch-PetSearch/blob/master/petsearch-shared/ui/home/build.gradle.kts)
[petsearch-shared:ui:petdetail](https://github.com/msa1422/KMM-Arch-PetSearch/blob/master/petsearch-shared/ui/petdetail/build.gradle.kts) 

KSP also requires to be implemented in
 [buildSrc build.gradle](https://github.com/msa1422/KMM-Arch-PetSearch/blob/master/buildSrc/build.gradle.kts)
Such a KSP setup is working fine in one of my projects which uses Androidx Room

These ViewModel use ```@NativeCoroutines```
[HomeViewModel](https://github.com/msa1422/KMM-Arch-PetSearch/blob/master/petsearch-shared/ui/home/src/commonMain/kotlin/com/msa/petsearch/shared/ui/home/HomeViewModel.kt)
[PetDetailViewModel](https://github.com/msa1422/KMM-Arch-PetSearch/blob/master/petsearch-shared/ui/petdetail/src/commonMain/kotlin/com/msa/petsearch/shared/ui/petdetail/PetDetailViewModel.kt)

Change version number here 
[libs.versions.toml : line51](https://github.com/msa1422/KMM-Arch-PetSearch/blob/master/gradle/libs.versions.toml)

I am unable to access fields marked with ```@NativeCoroutines``` on the iOS side.
I am relatively new to KMM. If I am doing anything wrong, please let me know.

**Comment by rickclephas** [comment]
Sorry for the delay! I have done some testing and for some reason the KSP plugin is never run. Which is why your declarations are disappearing.

This issue seems to be caused by the `kmm-shared-module-plugin`. Doing one of the following will "fix" the issue:
* Merging `kmm-shared-module-plugin` into your `build.gradle.kts` file
* Moving the KSP and KMP-NativeCoroutines plugins to your `kmm-shared-module-plugin`

**Comment by rickclephas** [comment]
So I am a little closer to the root cause and I am hoping to have a fix soon.
In the meantime please specify the KSP dependancy manually:
```kotlin
dependencies {
    add("kspIosX64", "com.rickclephas.kmp:kmp-nativecoroutines-ksp:1.0.0-ALPHA-7")
    add("kspIosArm64", "com.rickclephas.kmp:kmp-nativecoroutines-ksp:1.0.0-ALPHA-7")
    add("kspIosSimulatorArm64", "com.rickclephas.kmp:kmp-nativecoroutines-ksp:1.0.0-ALPHA-7")
}
```

**Comment by rezafaraji93** [comment]
> 

Where exactly should we add these dependencies?

**Comment by rickclephas** [comment]
These dependencies should be added to the top-level `dependencies` block in your build.gradle files that apply the KMP-NativeCoroutines plugin.
Also see the KSP docs [here](https://kotlinlang.org/docs/ksp-multiplatform.html).

**Comment by rickclephas** [comment]
FYI this issue has been fixed in [v1.0.0-ALPHA-8](https://github.com/rickclephas/KMP-NativeCoroutines/releases/tag/v1.0.0-ALPHA-8).

**Comment by msa1422** [comment]
Thanks. I appreciate your contribution. Code generation works like charm now.
